### ISSUE TITLE
SA1101 false negative when member is named same as class used for static invocation.

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/StyleCopAnalyzers#2211]
``` C#
public class Foo
{
    public Array Array { get; } = new int[0];

    public int IndexOf(object value) => Array.IndexOf(this.Array, value);
}
```

This shows up as violating SA1101 in the warnings list. When navigating to the actual code the warning goes away.

Wrote the repro code in the browser so not sure it compiles.

**Comment by bjornhellander** [comment]
Hej Johan!

I can confirm that the behaviour you're seeing is present in version 1.0.0.
Thank you for reporting it!

This behaviour seems to be caused by a combination of two problems. One bug in StyleCop which caused the diagnostic to be incorrectly reported in the first place and one in Roslyn which caused the diagnostic to be hidden when opening the file. The following pull requests are supposed to have solved them, but they are not yet part of a release:
#2097 - Fixed static class collisions in SA1101
#2033 - Make SA0000 hidden but enabled by default

I wrote "supposed to" above because an enabled SA0000 did not cause the diagnostic to be shown in open files for me when using 1.0.0. Don't know why...

But the original problem of ever reporting the diagnostic has at least been fixed by #2097 and that is the important part in this case. I tried your example in a unit test with the latest code and no diagnostics are reported anymore. Is it easy for you to work around the problem?

**Comment by JohanLarsson** [comment]
Yeah, it is not a problem living with it as it does not come up often. 
What decides when you release updated packages btw?

**Comment by bjornhellander** [comment]
@sharwell Can you answer this?

**Comment by pdelvo** [comment]
This looks like [https://github.com/dotnet/roslyn/pull/8167](this) could have fixed this. We should verify if this does not occur using the newest roslyn builds
